packages/salesforcedx-vscode-core/src/context/workspaceContext.ts

```diff
@@ -54,3 +63,136 @@ export class WorkspaceContext {
+  }
+
+  public async getConfigAggregator(): Promise<ConfigAggregator> {
+    const rootWorkspacePath = getRootWorkspacePath();
+    let configAggregator = this.configAggregators.get(rootWorkspacePath);
+    if (!configAggregator) {
+      configAggregator = await this.createConfigAggregator();
+      this.configAggregators.set(getRootWorkspacePath(), configAggregator);
+    }
+    return configAggregator;
+  }
+
+  public async getSfdxConfigAggregator(): Promise<ConfigAggregator> {
+    let sfdxConfigAggregator = this.sfdxConfigAggregators.get(
+      getRootWorkspacePath()
+    );
+    if (!sfdxConfigAggregator) {
+      sfdxConfigAggregator = await this.createConfigAggregator({ sfdx: true });
+      this.sfdxConfigAggregators.set(
+        getRootWorkspacePath(),
+        sfdxConfigAggregator
+      );
+    }
+    return sfdxConfigAggregator;
+  }
+
+  public async getGlobalConfigAggregator(): Promise<ConfigAggregator> {
+    if (!this.globalConfigAggregator) {
+      this.globalConfigAggregator = await this.createConfigAggregator({
+        globalValuesOnly: true
+      });
+    }
+    return this.globalConfigAggregator;
+  }
+
+  public async reloadConfigAggregators() {
+    console.log(
+      'The .sfdx config file has changed.  Reloading ConfigAggregator values in the salesforcedx-vscode-core package.'
+    );
+    // Force ConfigAggregator to load the most recent values from
+    // the config file.  This prevents an issue where ConfigAggregator
+    // can returned cached data instead of the most recent data.
+    const configAggregator = this.configAggregators.get(getRootWorkspacePath());
+    if (configAggregator) await configAggregator.reload();
+
+    const sfdx = this.sfdxConfigAggregators.get(getRootWorkspacePath());
+    if (sfdx) await sfdx.reload();
+  }
+
+  private async createConfigAggregator(
+    options: ConfigAggregatorOptions = {
+      sfdx: false,
+      globalValuesOnly: false
+    }
+  ): Promise<ConfigAggregator> {
+    let configAggregator;
+    const currentWorkingDirectory = process.cwd();
+    if (options.globalValuesOnly) {
+      ConfigAggregatorProvider.ensureProcessIsRunningUnderUserHomeDir(
+        currentWorkingDirectory
+      );
+    } else {
+      // Change the current working directory to the project path,
+      // so that ConfigAggregator reads the local project values
+      ConfigAggregatorProvider.ensureProcessIsRunningUnderProjectRoot(
+        currentWorkingDirectory
+      );
+    }
+    try {
+      configAggregator = options.sfdx
+        ? await SfdxConfigAggregator.create()
+        : await ConfigAggregator.create();
+    } finally {
+      // Change the current working directory back to what it was
+      // before returning.
+      // Wrapping this in a finally block ensures that the working
+      // directory is switched back to what it was before this method
+      // was called if SfdxConfigAggregator.create() throws an exception.
+      process.chdir(currentWorkingDirectory);
```

Comment:
do we need any error handling in a catch maybe?  Seems like a bad state if we encountered an error attempting to create an aggregator and it fails.

Reply:
There is at least one [place in plugin-source](https://github.com/salesforcecli/plugin-source/blob/b7829b99935b472833f8faf8652566fe446aa6b8/src/deployCommand.ts#L139-L140) where it is used and not wrapped in a t/c.  I think an exception would be rare and if there is one it should be thrown here.

Reply:
in the finally we won't know if there's an exception right?  So we'd need to catch it and save it then re-throw post chdir right?